src/components/Event/EventItem.tsx

```diff
@@ -1,118 +1,153 @@
-import { MouseEventHandler } from "react";
 import { EventItemStatus } from "./EventItemStatus";
 
-export interface iEventItem {
+export interface EventItemType {
   id: number;
   name: string;
   description: string;
-  startDateTime: string;
-  endDateTime: string;
+  startDateTime: Date;
+  endDateTime: Date;
   image?: StaticImageData;
   price?: number;
   maxParticipants?: number;
   inviteLink?: string;
-  address: string;
+  address?: string;
   lat?: number;
   lng?: number;
   status: EventStatus;
   isPrivate: boolean;
   isFree: boolean;
-  createdAt: string;
-  updatedAt?: string;
-  host: User;
-  eventParticipants: User[];
+  createdAt: Date;
+  updatedAt?: Date;
+  createdBy: User;
+  eventParticipants: EventParticipants[];
+}
+
+interface EventParticipants {
+  user: User
 }
 
-export interface iEvent {
-  event: iEventItem;
+const eventWithParticipants = Prisma.validator<Prisma.EventDefaultArgs>()({
+  include: { eventParticipants: { include: {user: true}}, createdBy: true },
+});
+export type EventWithParticipants = Prisma.EventGetPayload<
+  typeof eventWithParticipants
+>;
+
+export interface EventItemProps {
+  event: EventWithParticipants | EventItemType;
   onClick?: MouseEventHandler;
 }
 
-const EventItem = (props: iEvent) => {
+const EventItem = (props: EventItemProps) => {
   return (
-    <>
-      <div
-        className="cursor-pointer rounded-lg border shadow-md hover:shadow-xl"
-        onClick={props.onClick}
-      >
-        {props.event.image && (
-          <Image
-            className="rounded-t-lg"
-            src={props.event.image}
-            alt="Event Image"
-          />
-        )}
-        <div className="p-5 md:p-6 lg:p-10">
-          <div className="mb-8 flex flex-wrap items-center justify-between gap-5">
-            <div className="flex flex-wrap gap-1">
-              <div className="h-10 w-10 rounded-full border bg-black text-white">
-                <p className="mt-2 text-center">
-                  +{props.event.eventParticipants.length}
-                </p>
+    <div
+      className="cursor-pointer rounded-lg border shadow-md hover:shadow-xl"
+      onClick={props.onClick}
+    >
+      {props.event.image && (
+        <Image
+          className="rounded-t-lg w-full"
+          src={props.event.image}
+          alt="Event Image"
+        />
+      )}
+      <div className="p-5 md:p-6 lg:p-10">
+        <div className="mb-8 flex flex-wrap items-center justify-between gap-5">
+          <div className="flex flex-wrap gap-1">
+            <div className="h-10 w-10 rounded-full border bg-black">
+              <div className="mt-2 text-center text-white">
+                +{props.event.eventParticipants.length}
               </div>
-              {props.event.eventParticipants.map((participants) => {
+            </div>
+            {props.event.eventParticipants.map((participant) => {
                 return (
                   <>
```

Comment:
The usage of Array.map here requires the child component to have a key attribute.

Reply:
That is an error from the previous merge. I fixed here.